# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build xorg-server-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.14; lt-2.4.2)
libtoolize -c -f; ACLOCAL_PATH=/x11/share/aclocal aclocal; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig ./configure --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/lib/xorg/modules/drivers
install src/.libs/fbdev_drv.so $DESTDIR/x11/lib/xorg/modules/drivers

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/xf86-video-fbdev-build.pc << EOF
Name: xf86-video-fbdev
Version: 0.4.4
Description: framebuffer device video driver for the X server
Requires: $REQUIRES

exec=\\
/x11/lib/xorg/modules/drivers/fbdev_drv.so
EOF

exit

# patch
--- xf86-video-fbdev-0.4.4.orig/configure.ac
+++ xf86-video-fbdev-0.4.4/configure.ac
@@ -28,6 +28,7 @@
         [xf86-video-fbdev])
 AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR(.)
 
 # Initialize Automake

# source
https://x.org/releases/individual/driver/xf86-video-fbdev-0.4.4.tar.gz
